pkg/pillar/pubsub/publish.go

```diff
@@ -37,6 +39,7 @@ type Publication struct {
 	publishToDir bool // Handle special case of file only info
 	dirName      string
 	persistent   bool
+	parser       types.ParseFunc
```

Comment:
Can pubsub depend on types??
Ideally, pubsub should be independent of Pillar. What is the need to bring in types into pubsub?

Reply:
See comment above.